apps/src/blockly/addons/cdoXml.js

```diff
@@ -108,6 +109,44 @@ export function addMutationToMiniToolboxBlocks(blockElement) {
   blockElement.removeAttribute('miniflyout');
 }
 
+/**
+ * Adds a mutation element to a block if it is a behavior definition.
+ * CDO Blockly uses an unsupported method for serializing state
+ * where arbitrary XML attributes could hold important information.
+ * Mainline Blockly expects a mutator. The presence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ */
+export function addMutationToBehaviorDefBlocks(blockElement) {
+  if (blockElement.getAttribute('type') !== 'behavior_definition') {
+    return;
+  }
+  const mutationElement =
+    blockElement.querySelector('mutation') ||
+    blockElement.ownerDocument.createElement('mutation');
+  // Place mutator before fields, values, and other nested blocks.
+  blockElement.insertBefore(mutationElement, blockElement.firstChild);
```

Comment:
if the block element already has the mutationElement, will this add it again or do nothing?

Reply:
I wondered the same thing! It turns out that this is safe because `insertBefore` will first remove the element from the element if needed, before re-inserting it.

https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore